### PR TITLE
[NXP][zephyr] Migrate to NXP Zephyr 3.7 downstream release

### PR DESCRIPTION
**Pull request by axelnxp**
- Kconfig/CMake updates for 3.7
- Optimize some stack sizes to reduce RAM footprint
- Optimize net configuration to reduce ram footprint
- Doc update
- Remove overlays from platform folders
- Workflows docker image updated to latest (depends on #35599)

### CODE REVIEW

**Comment by axelnxp** [comment]
The CI cheks will fail until the docker file update PR is merged

File: src/crypto/PSASessionKeystore.cpp [hunk 1]
```diff
@@ -68,12 +68,14 @@ class HkdfKeyAttributes : public KeyAttributesBase
     HkdfKeyAttributes() : KeyAttributesBase(PSA_KEY_TYPE_DERIVE, PSA_ALG_HKDF(PSA_ALG_SHA_256), PSA_KEY_USAGE_DERIVE, 0) {}
 };
 
+#if CHIP_CONFIG_ENABLE_ICD_CIP
```
Comment (axelnxp) [inline src/crypto/PSASessionKeystore.cpp:71]: FYI @maciejbaczmanski , following your PR #34925 , it introduced a warning (as error in our build) due to this function being defined but not used, so I put it under the same flag that the method calling it.
I hope this is ok for you.

File: scripts/build/builders/nxp.py [hunk 2]
```diff
@@ -240,6 +240,9 @@ def generate(self):
             if 'ZEPHYR_NXP_BASE' not in os.environ:
                 raise Exception("ZEPHYR_NXP_BASE need to be set")
 
+            if 'ZEPHYR_TOOLCHAIN_VARIANT' in os.environ:
```
Comment (axelnxp) [inline scripts/build/builders/nxp.py:243]: @andy31415 FYI, this fixes the nxp zephyr examples in the vscode image. This variable is set by nrf env, and clashes with our own, so we just unset it.

File: src/crypto/PSASessionKeystore.cpp (hunk 1 above)
Comment (maciejbaczmanski) [reply src/crypto/PSASessionKeystore.cpp:71]: oh, right, I've added this helper later and must've forgotten about the guard... Thanks!

**Review by andy31415 — APPROVED** [review]

File: scripts/build/builders/nxp.py (hunk 2 above)
Comment (andy31415) [reply scripts/build/builders/nxp.py:243]: Note however that the intent of the build scripts is that I can do a `--dry-run` and see shell commands required to execute the build. So you are not allowed to do run python specific commands: in this case del of the environment (or in other cases file manipulation or python process execution).

Please add the logic to the `cmd` to be executed by the shell, even if ugly.

Comment (axelnxp) [reply scripts/build/builders/nxp.py]: Understood, I changed it by adding `unset ZEPHYR_TOOLCHAIN_VARIANT` in `cmd`.

**Review by andy31415 — APPROVED** [review]

**Review by chapongatien — APPROVED** [review]

**Pull request merged**